genienlp/data_utils/bootleg.py

```diff
@@ -33,108 +33,21 @@
 import numpy as np
 import torch
 import ujson
-from bootleg.end2end.bootleg_annotator import BootlegAnnotator
+from bootleg.end2end.bootleg_annotator import BootlegAnnotator as BootlegClass
 from bootleg.end2end.extract_mentions import extract_mentions
 from bootleg.run import run_model
 from bootleg.utils.parser.parser_utils import parse_boot_and_emm_args
 
-from .database_utils import is_banned
+from .database_utils import is_banned, reverse_bisect_left
 
 logger = logging.getLogger(__name__)
 
 
-def reverse_bisect_left(a, x, lo=None, hi=None):
```

Comment:
Why is this function not just called "binary_search"? What's with the comment about keeping it reverse-sorted? The function doesn't modify the input at all.

Reply:
This is binary_search for a reverse sorted list (i.e. descending order). I adopted the naming convention in `bisect` library. 
Yeah, the comment is misphrased. It should be: "Locate the insertion point for x in a to **maintain** reverse sorted order"